plutus-scb/src/Cardano/BM/Data/Tracer/Extras.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Cardano.BM.Data.Tracer.Extras(
+    mkObjectStr
+    , inst
+    , it
+    , con
+    , PrettyToObject(..)
+    ) where
+
+import           Cardano.BM.Data.Tracer                (ToObject (..), mkObject)
+import           Data.Aeson                            (ToJSON (..), Value (String))
+import           Data.HashMap.Strict                   (HashMap)
+import qualified Data.HashMap.Strict                   as HM
+import           Data.Text                             (Text)
+import           Data.Text.Prettyprint.Doc             (Pretty (..), defaultLayoutOptions, layoutPretty)
+import qualified Data.Text.Prettyprint.Doc.Render.Text as Render
+import           Plutus.SCB.Events.Contract            (ContractInstanceId, IterationID)
+
+mkObjectStr :: Text -> [(Text, Value)] -> HashMap Text Value
+mkObjectStr str rest =
+    mkObject $ ("string", String str) : rest
+
+inst :: ContractInstanceId -> (Text, Value)
```

Comment:
It seems weird to me that this isn't somehow handled generically? Can't we just derive instances that have the right field names?

Reply:
there doesn't seem to be a suitable class for this at the moment but I can try to write one

Reply:
I made `Cardano.BM.Data.Tracer.Extras` a bit more generic in c434c00c581121d90a16658e6c56c9914cb8fafa.

Reply:
Looks reasonable - can we also ask upstream whether they have any tricks for this or would like a patch? Not sure quite who to ask, but I'm sure we can find someone.

Reply:
https://github.com/input-output-hk/iohk-monitoring-framework/issues/580